### PR TITLE
Debian support

### PR DESCRIPTION
**Pull request by majst01**
untested, also updates gardener dependencies, drop our gardener fork.

### CODE REVIEW

**Comment by majst01** [comment]
modify controllerregistration and generate according to:
https://github.com/gardener/gardener-extensions/pull/580

**Comment by majst01** [comment]
@Gerrit91 once tested, this is ready to be merged.

File: example/controller-registration.yaml [hunk 1]
```diff
@@ -6,11 +6,13 @@ metadata:
 spec:
   resources:
   - kind: OperatingSystemConfig
-    type: metal
+    type: ubuntu
```
Comment (Gerrit91) [inline example/controller-registration.yaml]: Unfortunately, we overlap with the Gardener os-controller for ubuntu, which also wants to watch "ubuntu" types. This leads to the controllers going wild. We should rename this to `ubuntu-metal` or `ubuntu-ignition`, same for the debian image.

**Comment by Gerrit91** [comment]
Deleting the vendor directory is such an act of liberation. Great!

File: example/controller-registration.yaml (hunk 1 above)
Comment (majst01) [reply example/controller-registration.yaml]: Added a comment to upstream:
https://github.com/gardener/gardener-extensions/pull/580#issuecomment-578786683

**Pull request merged**